js/app.js

```diff
@@ -342,3 +414,14 @@ function checkURL(link) {
   }
   return false;
 }
+
+/**
+ * Updates the favourite property in tile objects
+ * @param {string} id - The data-id value refering to the tile object key, against a star icon
+ * @param {string} icon - The data-icon value refering to favourite status, against a star icon
+ */
+function favStatusUpdate(id, icon) {
+  let bool = icon === "true" ? false : true;
+  STATE.tiles[id].favourite = bool;
```

Comment:
Call this function parsing the string value ( favStatusUpdate(id, Boolean(favourite));)

Reply:
`/**
 * Updates the favourite property in tile objects
 * @param {string} id - The data-id value refering to the tile object key, against a star icon
 * @param {boolean} currFav - The current value refering to favourite status of a tile
 */
function favStatusUpdate(id, currFav) {
  STATE.tiles[id].favourite = currFav ? false : true;
  // STATE.tiles[id].favourite = !currFav;

  console.log(STATE.tiles[id]);
}`